### DISCUSSION TITLE
How to timeout the CANoe.start_measurement function when stuck?

### DISCUSSION THREAD

**Discussion by SamuelGnehm** [chaitu-ycr/py_canoe#32] [Q&A] [answered]
Hey There,
I'm currently using your library for a work assignment. I have the problem that when i try to start the measurement without a license or there is an error when starting the measurement the CANoe.start_measurement gets stuck in an infinite wait state where it waits for the measurement to start. I tried many many things from multithreading to multiprocessing and try exept statements to timeout this function but couldn't get anything to work properly. is there any timeout function built in in your library that i didn't see or any way to time out this function that you know of that work? I would appreciate any help. Im relatively new to progrmming. also if i brought up this issue the wrong way, im sorry i just joined github
best regards
Samuel

**Comment by chaitu-ycr** [comment]
Measurement start timeout feature not available for now. I will consider your request and let u know here once implemented.

**Comment by chaitu-ycr** [comment, marked as answer]
@SamuelGnehm 

implemented this feature in latest release 2.1.1
upgrade to latest package ```pip install py_canoe --upgrade```

doc --> https://chaitu-ycr.github.io/py_canoe/999_reference/#src.py_canoe.CANoe.start_measurement
github release tag --> https://github.com/chaitu-ycr/py_canoe/tree/v2.1.1
